tests/jax_jit_test.py

```diff
@@ -205,6 +205,26 @@ def f(a, b, c):
     jitted_f = jax.jit(f)
     self.assertEqual(inspect.signature(f), inspect.signature(jitted_f))
 
+  def test_jit_compile_vmap_on_gpu(self):
+    if jtu.device_under_test() == "cpu":
+      self.skipTest("Test only runs on GPU")
+
+    @jax.vmap
+    def fn(x):
+      R1 = jnp.array([[x[0], 0, 0],
+                      [0, x[0], 0],
+                      [0, 0, x[0]]])
+      R2 = jnp.array([[x[0], 0, 0],
+                      [0, x[1], 0],
+                      [0, 0, x[2]]])
+      H = jnp.eye(4)
+      H = H.at[:3, :3].set(R2.T)
+      pos = H @ jnp.concatenate([x, jnp.array([1.0])])
+      return pos, R1
+    jitted_fn = jax.jit(fn)
+    x_gpu = jax.device_put(jnp.zeros((2,3)), jax.local_devices()[0])
```

Comment:
No need for explicit device put I think: when the tests are run on GPU, the default device is GPU.